docs/troubleshooting/Performance.mdx

```diff
@@ -79,7 +79,120 @@ module.exports = {
 
 See [Glob pattern in parser's option "project" slows down linting](https://github.com/typescript-eslint/typescript-eslint/issues/2611) for more details.
 
-## The `indent` / `@typescript-eslint/indent` rules
+## Changes to `extraFileExtensions` with `projectService` in your ESLint options
+
+When [`extraFileExtensions`](../packages/Parser.mdx#extrafileextensions) changes while using
+[`projectService`](../packages/Parser.mdx#projectservice), the TypeScript server must perform a full project reload.
+Project reloads can be observed using the [debug environment variable](../packages/typescript-estree/#debugging): `DEBUG='typescript-eslint:typescript-estree:*'`.
+
+```
+typescript-estree:useProgramFromProjectService Updating extra file extensions: before=[]: after=[ '.vue' ]
+typescript-estree:tsserver:info reload projects.
+typescript-estree:useProgramFromProjectService Extra file extensions updated: [ '.vue' ]
+...
+typescript-estree:useProgramFromProjectService Updating extra file extensions: before=[ '.vue' ]: after=[]
+typescript-estree:tsserver:info reload projects.
+typescript-estree:useProgramFromProjectService Extra file extensions updated: []
+...
+typescript-estree:useProgramFromProjectService Updating extra file extensions: before=[]: after=[ '.vue' ]
+typescript-estree:tsserver:info reload projects.
+typescript-estree:useProgramFromProjectService Extra file extensions updated: [ '.vue' ]
+```
+
+As files are received, if `extraFileExtensions` differs from the previous file,
+the TypeScript server is updated to match your ESLint options.
+The TypeScript server incrementally updates the project structure by creating an execution pipeline.
```

Comment:
[Docs] I think we can get away without mentioning the concept of the execution pipeline...

```suggestion
```